src/measure_theory/conditional_expectation.lean

```diff
@@ -262,4 +273,88 @@ end complete_subspace
 
 end Lp_meas
 
+/-! ## Conditional expectation in L2
+
+We define a conditional expectation in `L2`: it is the orthogonal projection on the subspace
+`Lp_meas`. -/
+
+section condexp_L2
+
+local attribute [instance] fact_one_le_two_ennreal
+
+variables [complete_space E] [borel_space 𝕜] {m m0 : measurable_space α} {μ : measure α}
+  {s t : set α}
+
+local notation `⟪`x`, `y`⟫` := @inner 𝕜 E _ x y
+local notation `⟪`x`, `y`⟫'` := @inner 𝕜 E' _ x y
+local notation `⟪`x`, `y`⟫₂` := @inner 𝕜 (α →₂[μ] E) _ x y
+
+variables (𝕜)
+/-- Conditional expectation of a function in L2 with respect to a sigma-algebra -/
+def condexp_L2 (hm : m ≤ m0) : (α →₂[μ] E) →L[𝕜] (Lp_meas E 𝕜 m 2 μ) :=
```

Comment:
I wonder if `m` should be explicit in this definition. The reason is that, in the goal view, `condexp_L2 𝕜 hm f` will show as `condexp_L2 𝕜 _ f`, so there is no way to guess which sigma-algebra one is speaking about. When one starts speaking of martingales with a lot of different sigma-algebras, this could be a mess. (Of course, I don't expect we will use `condexp_L2` a lot, the `L1` version should be more important, but the same question will show up there).

Reply:
Yes, condexp_L2 is a tool to build the "true" conditional expectation, and will then be forgotten. But I see your point and will make `m` explicit.

Reply:
Wait. It actually does show as `condexp_L2 𝕜 hm f`. I can't find a goal with a `_` instead of `hm`.

Reply:
That's great. Then no need to change anything!